packages/core/src/provider/base.ts

```diff
@@ -537,6 +536,19 @@ export class SmartAccountProvider<
     ) => TAccount
   ): this & { account: TAccount } => {
     const account = fn(this.rpcClient);
+
+    // sanity check. Note that this check is only performed iff the optional entryPointAddress is given upon initialization.
```

Comment:
only because some people might think it's a typo and not the proof notation shorthand 😅 
```suggestion
    // sanity check. Note that this check is only performed if and only if the optional entryPointAddress is given upon initialization.
```